Favicon and Suggestions

whippingdot opened the pull request:
This is the pull request to make the suggestions page work and the favicon come up on the website.

whippingdot commented on index.js:
```diff
@@ -3,10 +3,11 @@ const app = express();
 
 app.get("/", async(req, res) => {
-  console.log("Yeeet, [DEBUG] we running in home page");
+  console.log(`${process.env.REPL_OW}: Yeeet, [DEBUG] we running in home page`);
```
WHAT

whippingdot commented:
hey the favicon isn't loading for me anymore. Can anyone check if it works on both pages. @SixBeeps @EpicRaisinGaming

whippingdot commented on index.js:
```diff
@@ -1,30 +1,61 @@
-app.get("/", async(req, res) => {
-  console.log("Yeeet, [DEBUG] we running in home page");
-  res.sendFile(`${__dirname }/views/home.html`);
+app.get("/", async (req, res) => {
```
No it is suposed to be like that, continuation on next line

firefish111 commented:
who is this hound guy

firefish111 commented on index.js:
```diff
@@ -1,30 +1,66 @@
+
+app.use("favicon.ico", express.static(__dirname + "/favicon.ico"));
+
+app.get("/", async (_, res) => {
```
very useful i totally know exactly which of those lines you mean

whippingdot replied:
TRUE. But if you use vs code or something you will no exactly which line.

whippingdot replied:
@firefish111 this is also wrong as it doesn't see the lines below, ignore it unless you see something meaningful

firefish111 replied:
> TRUE. But if you use vs code or something you will know exactly which line.

haha vim go brrr

firefish111 commented on index.js:
```diff
@@ -1,30 +1,66 @@
+app.use("favicon.ico", express.static(__dirname + "/favicon.ico"));
+
+app.get("/", async (_, res) => {
+	console.log(`Signing in [DEBUG]`);
```
is there a way to make you use ES2020 by default

firefish111 replied:
because nodejs 16 supports ES2020

firefish111 commented:
```js
app.use("favicon.ico", express.static(__dirname + "/favicon.ico"));
```
Excuse me the favicon is done from html
```html
<link rel="icon" href="./favicon.ico" />
```

firefish111 commented:
# Remade favicon as SVG in higher quality
# As mentioned above favicon does not have to be an `ico` for html
![replit](https://storage.googleapis.com/replit/images/1609588547078_c2ad5c0056e3db39eab515cf9d536611.svg)

whippingdot commented:
> ```js
> app.use("favicon.ico", express.static(__dirname + "/favicon.ico"));
> ```
> 
> # Excuse me the favicon is done from html
> ```
> <link rel="icon" href="./favicon.ico" />
> ```
> 
> and it doesn't have to be an `ico` it can be a `png`, `jpg`, or even `svg`

Bro we finished the favicon yesterday. Nice work and thank you but it was done...I think. I worked with @SixBeeps and @EpicRaisinGaming to fix it yesterday

whippingdot commented:
> # Remade favicon as SVG in higher quality
> # As mentioned above favicon does not have to be an `ico` for html
> ![replit](https://camo.githubusercontent.com/891f88d75932c850d9bea1f399e25dae7b75c6e66167c279f11a7a17d34ce89c/68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f7265706c69742f696d616765732f313630393538383534373037385f63326164356330303536653364623339656162353135636639643533363631312e737667)

OH MY!. I didn't know SVGs had such good quality. I will save that pic and make it as our group logo.!! 😮 😮

whippingdot commented:
@firefish111 Now we have some new things to do. Favicon is done. Suggestions output in the console, but we can't be hawk watching the console all the time(@AmazingMech2418) so we need to output the suggestions in a text file/DB. We have to make sure that each person is only allowed 1 suggestion too(for the next release below). After that, we can implement the suggestions into the main branch/shared project as it is a stable release(I will re-share).

Then, we can work on making the suggestions pop up below the form for giving the suggestions. This means that below the form for inputting suggestions, there will be a list of all suggestions inputted by people. That is the reason why only one suggestion should be allowed per person. Then we can release that version too. That will be the end of the project. I have more ideas, but then we will be working on this forever, so this is enough. We have to do other projects later too right?

firefish111 commented:
svgs are scalable, they can be any quality you like, that's the idea @whippingdot

firefish111 commented:
@whippingdot Sorry, you can't put svgs as pfps.
# JUST FOR THE PFP, NOT FOR THE FAVICON: here is 2000x2000px png
![replit](https://user-images.githubusercontent.com/33234914/103459132-caa5bf00-4d04-11eb-95df-ba242f22f58f.png)

whippingdot commented:
Ok cool, I will add it rn. I haven't added it yet.

AmazingMech2418 commented:
@whippingdot Are we working on this here or on the repl that is the 2021 version? By the way, I don't think we should push this PR since we already have favicons working... Are we planning Repl Auth though? Also, what DB system are we thinking of? ReplDB?

firefish111 commented:
@AmazingMech2418 REPLDB is unreliable, and can be broken into, ~~as my old trusty tutorial told us~~.
Try writing our own? Because JSON writing is ~~un~~reliable as well (actually if @mat1's repl reviver works then it should work!)

AmazingMech2418 commented:
@firefish111 
> @AmazingMech2418 REPLDB is unreliable, and can be broken into, ~as my old trusty tutorial told us~.
> Try writing our own? Because JSON writing is ~un~reliable as well (actually if @mat1's repl reviver works then it should work!)

Remember, Repl.it doesn't have file persistence...

firefish111 commented:
# READ THE PARENTHESES REEEEE @AmazingMech2418

whippingdot commented:
> @whippingdot Are we working on this here or on the repl that is the 2021 version?

You can work wherever you want. You can even push changes. I realized the repl shared on repl talk can only change if I pull changes. I am not going to do that so you can push changes if you want. Just try to work on the repl I invited you to, but you don't have to.

> By the way, I don't think we should push this PR since we already have favicons working...

This pull request is not only for favicons. Did you read what I said above:
> @firefish111 Now we have some new things to do. Favicon is done. Suggestions output in the console, but we can't be hawk watching the console all the time(@AmazingMech2418) so we need to output the suggestions in a text file/DB. We have to make sure that each person is only allowed 1 suggestion too(**for the next release below**). After that, we can implement the suggestions into the main branch/shared project as it is a stable release(I will re-share).

We will push close this pr and merge changes when we are done with the suggestions. Then I will re-share. Then we will create another pr for more changes.

> Are we planning Repl Auth though? Also, what DB system are we thinking of? ReplDB?

Sorry but I don't know about DBs and all. I think @firefish111 answered your question anyway. We in fact **are** doing the repl auth, and I started working on it yesterday. If you saw the project, you would know. I don't know how to re-direct the page when done authenticating though so I haven't finished. Also I don't know how to restrict the pages. For now people can just put a /home at the ending of the link and it will work.

whippingdot commented:
I don't know if anyone will actually see this, but I hope you do. This is the plan:
Repl Auth: @whippingdot @firefish111 @rohilpatel1 @Isaiah08-D @ColePete 
Suggestions send to file/DB/JSON: @AmazingMech2418 @SixBeeps @EpicRaisinGaming @cursorweb @Codemonkey51 

I mentioned a lot of people for these projects cus half of you guys will do nothing. Just try to help each other out and finish this. @AmazingMech2418 cause you are the manager if you want you can add to this and tell who to do what...

firefish111 commented:
I can do auth, I've done it twice before. Is it on the feature branch? Or on master? @whippingdot

whippingdot commented:
> Is it on the feature branch? Or on master? @whippingdot

We are doing everything on the feature branch. For some reason the CSS isn't working for the replAuth page, you can look at what I have done by now. I have done a lot, but some don't work. You can finish the Repl Auth if you want. There is some stuff to do after that, but I will do it. One thing to note is that after signing in, we need the page to be redirected to /home. We also need it to know if you have already signed/loged in so you don't have to sign in again each time.

Isaiah08-D reviewed (approved):
Looks fine, i agree to it.

whippingdot commented:
> Looks fine, i agree to it.

Noooo we still have to finish the suggestions. This is not done yet. We still have to finish repl auth and suggestions in this pr.

firefish111 commented:
@whippingdot Why redirect? Can't we just use a templating engine, for instance pugjs? ~~ejs is rubbish, lets forget it exists~~

firefish111 commented:
And why are the upvoted posts in concrete on the website? Can't we send a GraphQL query towards it? ~~ask rayhanadev i have no idea how to do that~~

firefish111 commented on index.js:
```diff
@@ -1,30 +1,66 @@
-app.use('favicon.ico', express.static(__dirname + '/favicon.ico'))
+app.use("/static", express.static(__dirname + "/static"));
+
+app.use("favicon.ico", express.static(__dirname + "/favicon.ico"));
```
I hope you fixed line 23 now

whippingdot replied:
Um... we have so many trial and errors in the code I don't know which one even does anything. As for the suggestions we have some code in index.js and we have some code in the HTML itself, which one is the right code and which one does something is a mystery.

whippingdot replied:
I will do it

firefish111 commented on views/home.html:
```diff
@@ -9,17 +9,17 @@
-    />  
+		<link
+			rel="shortcut icon"
+			href="https://Replit-2020.whippingdot.repl.co/static/favicon_but_better.ico"
```
use relative `./` not absolute

whippingdot replied:
Maybe, but it doesn't work. Did you forget the hours we used to try fixing this. It was our first problem

firefish111 replied:
It should be `rel="icon"` @whippingdot

whippingdot replied:
Try it yourself on the repl. Also @BD103 @JBYT27 @zplusfour and @darkdarcool joined the org. but @HuongTuon(Wumi4) left.

firefish111 commented on views/replAuth.html:
```diff
@@ -0,0 +1,26 @@
+		<h1>Log In</h1>
+		<h2>Log in using your repl.it account!</h2>
+		<script
+			authed="alert('You Logged In')"
```
yes. it needs to refresh, and just throw in an if statement in pugjs or something. I'll get round to converting.

whippingdot replied:
Read my and @AmazingMech2418 's comments. We might ditch this page altogether and make it just a button on the suggestions page. Sorry...

whippingdot replied:
@firefish111 I like pugjs and all but it is confusing. To keep it simple let us not use pugjs here.

firefish111 replied:
pugjs is YAML-like, so if you know YAML it should be easy to use.
YAML (YAML Ain't a Markup Language)  basically is fancy indent-reliant json.
For example: (from google):
```yml
---
- hosts: webservers

  vars:
    http_port: 80
    max_clients: 200

  remote_user: root

  tasks:
  - name: ensure apache is at the latest version
    yum:
      name: httpd
      state: latest

  - name: write the apache config file
    template:
      src: /srv/httpd.j2
      dest: /etc/httpd.conf
    notify:
    - restart apache

  - name: ensure apache is running
    service:
      name: httpd
      state: started

  handlers:
    - name: restart apache
      service:
        name: httpd
        state: restarted
```
It is heavily indent-reliant, much like Python and Nim. That means, one incorrect indent, and your program might not behave as expected.

whippingdot replied:
K...But most of us don't use pug. I can create a team under web dev for pugjs if you want but this project will remain strictly JS, CSS, HTML, and Node.

firefish111 commented on views/suggestions.html:
```diff
@@ -80,7 +79,7 @@ <h1>Share It</h1>
 		<h1 class="center" id="heading">Suggest something about Repl.it!</h1>
 		<div id="form" class="center">
-			<form>
+			<form action="/api/suggestions" method="POST">
```
I would say this is bad practice, but I have no idea how actions work so I have a workaround using simple js

whippingdot replied:
Why is this bad practice. I am not done learning JS yet...Or CSS

firefish111 replied:
It's not, but I have no idea how `action` works and the docs don't explain very well. I think it performs a `PUT` request against the uri provided, but I have no idea.

firefish111 commented on views/timeline.html:
```diff
@@ -9,18 +9,18 @@
 		<header class="center">
 			<h1 class="big">For Repl.it</h1>
 			<h1>Share It</h1>
 			<div class="navbar">
```
navbar class? so what is the `nav` element for?

whippingdot replied:
I didn't even know that was a thing...Anyway someone else did that, not me. Just leave it be, makes no difference. If you want you can change it.

firefish111 commented:
and text is very hard to read lemme fix

ColePete commented:
Dang, I am inactive for a week because of Christmas and New Years' and I miss sooooooo much. Can someone fill me in?? Also, @whippingdot what is an auth?? I can do it, I just need to know what it is.

firefish111 commented:
@ColePete 
![image](https://user-images.githubusercontent.com/33234914/103486774-28670380-4df8-11eb-8d39-46eefcbd1848.png)
![image](https://user-images.githubusercontent.com/33234914/103486788-446aa500-4df8-11eb-98ec-3a842dcb41a3.png)

AmazingMech2418 commented:
I fixed Repl Auth. Still a little work to do though... Personally, I think it'd be best to make the auth in a login page and suggestions require you to log in, but everything else be in the old locations. Like home page at / instead of /home.

firefish111 reviewed (changes requested):
# Already reviewed this?!?

AmazingMech2418 commented:
> # Already reviewed this?!?

What?

firefish111 commented:
@AmazingMech2418 Clicking random buttons.
![image](https://user-images.githubusercontent.com/33234914/103490010-d088c680-4e10-11eb-9190-ad003c784836.png)

programmeruser2 commented on index.js:
```diff
@@ -1,30 +1,66 @@
+	res.sendFile(`${__dirname}/views/suggestions.html`);
+});
+
+app.post("/api/suggestions", require("body-parser")(), (req, res) => {
```
The `body-parser` package is not needed here as Express now comes with a body parser (`express.urlencoded` and `express.json`).

programmeruser2 commented on index.js:
```diff
@@ -1,30 +1,66 @@
 
 // Finally got this line to work, so /static leads to any necessary static content. ~ SixBeeps
-app.use('/static', express.static(__dirname + '/static'));
-app.use('favicon.ico', express.static(__dirname + '/favicon.ico'))
```
Why is this here? Can't you just move the favicon into the static folder?

whippingdot replied:
I don't know, there were some glitches and stuff wasn't working. You know you and @firefish111 and anyone really can change the project however they want. Just make sure you revert changes if something doesn't work. Stop requesting changes. Also don't review changes, review the changes when this pr is done, we aren't done with the pr yet.

firefish111 replied:
@whippingdot You can do some comment isolation.
Basically comment out some things that people say won't work, and if it doesn't work, then uncomment that line out a comment something else around it. It's a very good strategy for bug-fixing.

whippingdot replied:
Ok, thanks for the help

whippingdot commented:
> I fixed Repl Auth. Still a little work to do though... Personally, I think it'd be best to make the auth in a login page and suggestions require you to log in, but everything else be in the old locations. Like home page at / instead of /home.

bro, are you a different version of me or something. I was thinking the same thing yesterday night but I didn't have internet so I couldn't push changes. I changed the ROADMAP.md to say revert everything back to normal and in the suggestions page make it so the repl auth button is there under the name. I want to do this but it seems someone has put a lot of work into the page already. I need confirmation from at least 2 people to do this. Should I do this @Repl-it-Coders/web-dev

whippingdot commented:
> Dang, I am inactive for a week because of Christmas and New Years' and I miss sooooooo much. Can someone fill me in?? Also, @whippingdot what is an auth?? I can do it, I just need to know what it is.

Repl auth was finished by @firefish111 and @AmazingMech2418 so you don't have to do anything. I shared the website, check it out. We forked the repl and started making new changes as we didn't finish the website earlier.

whippingdot commented:
> And why are the upvoted posts in concrete on the website? Can't we send a GraphQL query towards it? ~~ask rayhanadev i have no idea how to do that~~

First of all, the posts there are the most upvoted posts **of 2020** and not all time. Only posts posted in 2020 are allowed. Second of all, it is WAY easier to do manually and get the most upvoted posts(we don't need to do anything now cause 2020 is over).

whippingdot commented:
Ok, cause everyone is sleeping I am going to push all the changes made which ended up with repl auth, and then I am going to call it "revert here if wanted", and then I am going to delete everything and change everything so it comes back to normal and I am going to try to make the repl auth a button underneath "name" in suggestions.

whippingdot commented:
Guys, I found a mistake. We forked the repl from the old repl right, so all the css links and everything was linking to the old repl. We/I am soooo dumb. That is why the CSS changes I was making wasn't working. That is why the favicon wasn't showing. LOOOOL. If you find something linked to 2020 please change it to 2021.

firefish111 commented:
> Guys, I found a mistake. We forked the repl from the old repl right, so all the css links and everything was linking to the old repl. We/I am soooo dumb. That is why the CSS changes I was making wasn't working. That is why the favicon wasn't showing. LOOOOL. If you find something linked to 2020 please change it to 2021.

This is why you use relative paths.

whippingdot commented:
Well there was a problem earlier with relative paths. If you want try to make it relative paths. Code in the repl I invited you to.

whippingdot commented:
Hey @BD103, @JBYT27, @zplusfour, and @darkdarcool try to read the comments here and you will get a gist of what is going on. We finished the project but not completely so we forked the repl and in the shared one we are not pulling changes. We are trying to add more stuff and when done we will re-share on repl talk. The features we need to add are suggestions being sent to a file/DB/JSON file and the things we need to change is make repl auth underneath name in the suggestions page, remove the box for name in the suggestions page, and make it so the js takes you name after authenticating so we can use that for suggesting. No-one is allowed to suggest more than once too(because of future update)

darkdarcool commented:
@whippingdot I get what your saying, but where is the name tag that you want to move the repl auth to? I tried to find it but couldn't. Can you let me know where it is?

whippingdot commented:
> @whippingdot I get what your saying, but where is the name tag that you want to move the repl auth to? I tried to find it but couldn't. Can you let me know where it is?

![image](https://user-images.githubusercontent.com/64354499/103542186-94ae3980-4ec2-11eb-8297-58bc162e4c85.png)
username...name...samething

darkdarcool commented:
Sorry @whippingdot, it early for me and I'm still waking up.

darkdarcool commented:
I meant where was the file. And another thing, can i have the link so I can visualize what I'm doing please?

whippingdot commented:
> I meant where was the file. And another thing, can i have the link so I can visualize what I'm doing please?

Ohhh... @BD103 and @JBYT27 and @zplusfour and @darkdarcool make changes in the repl I invite you to. I forgot inviting you guys and I will do it now.

darkdarcool commented:
Thanks @whippingdot , will get there as soon as possible, hope to talk to you there.

AmazingMech2418 commented:
> > @AmazingMech2418: I fixed Repl Auth. Still a little work to do though... Personally, I think it'd be best to make the auth in a login page and suggestions require you to log in, but everything else be in the old locations. Like home page at / instead of /home.
> 
> bro, are you a different version of me or something. I was thinking the same thing yesterday night but I didn't have internet so I couldn't push changes. I changed the ROADMAP.md to say revert everything back to normal and in the suggestions page make it so the repl auth button is there under the name. I want to do this but it seems someone has put a lot of work into the replAuth page. So, I will push changes now and then make new changes. We can revert back the changes I made if you want when you wake up.

Well, I did  the replAuth2 page to change it some...

ColePete commented:
Can someone resend the link plz?

darkdarcool commented:
To what @ColePete ?

ColePete commented:
> To what @ColePete ?

The repl

darkdarcool commented:
Ok, do have anything to work on in there?
https://repl.it/@Whippingdot/Replit-2021?from=notifications#views/replAuth2.html

ColePete commented:
Isn't that what everybody is working on?

darkdarcool commented:
Yes everyone in the web dev team is, but what are you working on specifically

ColePete commented:
I am in the web dev team

darkdarcool commented:
What was your assignment in the repl?

ColePete commented:
I don't remember, where can I find that out?

darkdarcool commented:
Can you work to fix the repl auth?
I'm not one to assign, @whippingdot is, but we do need help on the auth, can you work on that?

ColePete commented:
Sure, what needs to be fixed?

darkdarcool commented:
The repl.it auth

ColePete commented:
thanks, what do you want it to do?

darkdarcool commented:
The auth works, but it dosen't take you anywhere when you finish logging in, can you make it go ot the home page?

darkdarcool commented:
WORK IN THE REPl, NOT GITHUB

ColePete commented:
Sure, I am in class right now, but when I finish, will do.

darkdarcool commented:
kk

AmazingMech2418 commented:
> The auth works, but it dosen't take you anywhere when you finish logging in, can you make it go ot the home page?

Um... It works completely...

darkdarcool commented:
It works here for me it doesn't try again with the one that we are woking on.
https://replit-2021.whippingdot.repl.co/

AmazingMech2418 commented:
> It works here for me it doesn't try again with the one that we are woking on.
> https://replit-2021.whippingdot.repl.co/

It was working before... Did anyone change anything?

ColePete commented:
So, do I need to fix it or nah?

AmazingMech2418 commented:
Someone removed the JS code... :/

AmazingMech2418 commented:
Wait, never mind. Someone removed the CSS though...

AmazingMech2418 commented:
Also, we really need to fix the CSS... Too little contrast and way too bright...

ColePete commented:
agreed

darkdarcool commented:
#Hi
So @AmazingMech2418 , @headiscoding and I have made the site look better!
Like the repl auth works good
Styling the site
Made it so when you auth you auto gat your name put in on the suggestion page when you type.
I want to push changes but need your guys OK first to push in the feature branch.
Give me the ok if it looks good!

firefish111 commented:
oh no so many comments i cannot comprehend

darkdarcool commented:
What happened the repl changes that we made today are gone?

Isaiah08-D reviewed (approved):
Looks great! I approve.

darkdarcool commented:
Thx @Isaiah08-D

whippingdot commented:
> Can you work to fix the repl auth?
> I'm not one to assign, @whippingdot is, but we do need help on the auth, can you work on that?

No, @AmazingMech2418 is also the one to assign

whippingdot commented:
> Looks great! I approve.

WE AREN'T DONE YET. WE HAVE A MIILLION PULL REQUESTS. thanks for your feedback tho @Isaiah08-D . I will finish and maybe share today itself. I have a lot of work to do

whippingdot commented:
UMMMM I have some things to note
1. WHO IN THE WORLD CHANGED THE MAIN BRANCH??????? It says there are merge conflicts. Are you **trying** to give me work???
2. Why when I wake up are there a million prs. @darkdarcool please cool down and stop creating prs. All prs have the same description and the same name. Next time when you create a pr explain what the changes are properly. I don't understand what they are. You also closed them without saying anything. If you create a pr and you want the changes to take effect you have to merge it.
3. @darkdarcool if you create a pr and say that you are going to wait for me to close/merge it then why do you merge it without me. @BD103 and @darkdarcool try to wait for my approval next time... I want to know what the changes are
4. Someone messed up the whole website. The navbar is BROKEN.

I am going to fix all the mistakes you made and I am going to change everything until there are only two branches. Thanks, everyone for the changes you made but next time wait for me to come before merging the prs. I wanna see the changes.

BTW I had more than 100 notifications yesterday. LES GO!! the community is active. I thank everyone who is helping with the changes. I am not telling you to stop making changes, all I am saying is if you create a pr, have a proper description, if you want to merge a pr, wait for my approval, and if you want to make the changes from the pr come into the project, ___merge___ the pr, do not close it.

whippingdot commented:
I found out who it was, @darkdarcool made changes to made. I am rebasing right now.

darkdarcool commented:
@whippingdot , Sorry about the pull request, but I made a mistake. Everything was supposed to go to feature and I thought  requested a review, all I know I wasn't online for most of what you just said.

whippingdot commented:
> Hey. Sorry for not waiting, @whippingdot. So are you going to be the one who manages the conflicts? I tried to do that, but I honestly am not good around JS.

It is ok, but if you make major changes in a pr, wait for me next time. I can try to fix the conflicting files as soon as I can but really, it doesn't matter now as we still have stuff to do before merging.

whippingdot commented:
Also @BD103 and @darkdarcool I am kinda acting like the boss here, and I don't want you guys to get that feeling. You can do whatever you want, but I just wanted to see the changes too and I would like to have an opinion. I can't really do that that well unless you wait for me as I am sleeping while you guys are working.

BD103 commented:
> Also @BD103 and @darkdarcool I am kinda acting like the boss here, and I don't want you guys to get that feeling. You can do whatever you want, but I just wanted to see the changes too and I would like to have an opinion. I can't really do that that well unless you wait for me as I am sleeping while you guys are working.

It honestly doesn't feel like that, actually, so you're fine. Plus, your the owner / admin of this group, so its your job to keep an eye on things. Don't worry about it! 😄

darkdarcool commented:
> Also @BD103 and @darkdarcool I am kinda acting like the boss here, and I don't want you guys to get that feeling. You can do whatever you want, but I just wanted to see the changes too and I would like to have an opinion. I can't really do that that well unless you wait for me as I am sleeping while you guys are working.

Your 100% fine! I too would like to have some say and an opinion on things like these.

darkdarcool commented:
Ok. I was looking at some of the communities, and a lot of them are bare. So I had a proposition.
I have been working a java text adventure for the java community, and I wanted to know if it was OK to add?
This is just an idea to get more activity on other branches of this community, so it is ok to shoot this idea down.  
Tell me what you think!

BD103 commented:
> Ok. I was looking at some of the communities, and a lot of them are bare. So I had a proposition.
> I have been working a java text adventure for the java community, and I wanted to know if it was OK to add?
> This is just an idea to get more activity on other branches of this community, so it is ok to shoot this idea down.
> Tell me what you think!

I like the idea, but it's not my place to say yes or no. But, could you possibly say this somewhere else? Not in an open PR?

darkdarcool commented:
Kk

whippingdot commented:
Sorry I am responding here but yes, you can create a project for the Java team. One thing tho. I don't want this org to have a million repos and I want it to be clean. I don't know how to do that without telling everyone to not make repos here, even tho I want you to. You can create repos here, but I want some kind of order. @AmazingMech2418 is already a mod, @firefish111 also has powers where he answers discussions/questions by org members, so the person I think can do this with me(create some kind of order) is @BD103. Soon @darkdarcool will also get something, but for now @BD103 you have to help me create some sense of order to the creation of our repos.

BD103 commented:
> Sorry I am responding here but yes, you can create a project for the Java team. One thing tho. I don't want this org to have a million repos and I want it to be clean. I don't know how to do that without telling everyone to not make repos here, even tho I want you to. You can create repos here, but I want some kind of order. @AmazingMech2418 is already a mod, @firefish111 also has powers where he answers discussions/questions by org members, so the person I think can do this with me(create some kind of order) is @BD103. Soon @darkdarcool will also get something, but for now @BD103 you have to help me create some sense of order to the creation of our repos.

So you want me to organize and moderate the repos based on a set of guidelines created by both of us? I down for that.

whippingdot commented:
> > Sorry I am responding here but yes, you can create a project for the Java team. One thing tho. I don't want this org to have a million repos and I want it to be clean. I don't know how to do that without telling everyone to not make repos here, even tho I want you to. You can create repos here, but I want some kind of order. @AmazingMech2418 is already a mod, @firefish111 also has powers where he answers discussions/questions by org members, so the person I think can do this with me(create some kind of order) is @BD103. Soon @darkdarcool will also get something, but for now @BD103 you have to help me create some sense of order to the creation of our repos.
> 
> So you want me to organize and moderate the repos based on a set of guidelines created by both of us? I down for that.

One thing, again, is that in a team you can create repos. Make sure that projects not done by like half of the whole org are stored in the corresponding language's team

ColePete commented:
> One thing, again, is that in a team you can create repos. Make sure that projects not done by like half of the whole org are stored in the corresponding language's team

so, lets get started.

BD103 commented:
> One thing, again, is that in a team you can create repos. Make sure that projects not done by like half of the whole org are stored in the corresponding language's team

Are you sure? I tried that, and it didn't work.

whippingdot commented:
@ColePete ???

whippingdot commented:
> > One thing, again, is that in a team you can create repos. Make sure that projects not done by like half of the whole org are stored in the corresponding language's team
> 
> Are you sure? I tried that, and it didn't work.

I responded somewhere else

ColePete commented:
> @ColePete ???

What?

whippingdot commented:
what are you saying
> so, lets get started.


for?

ColePete commented:
i thought that we were going to start a new repl. Sorry I am drowning in how active you guys are with sooooooo many messages.

whippingdot commented:
> i thought that we were going to start a new repl. Sorry I am drowning in how active you guys are with sooooooo many messages.

No, we are not creating a new repl. Also you just have to look at the most recent messages, if you don't understand or you want more context you can scroll back.

darkdarcool commented:
@whippingdot , I just got on so do you have anything for me to do today on the repl today?

whippingdot commented:
> @whippingdot , I just got on so do you have anything for me to do today on the repl today?

We are editing the branch that @BD103 created. Make it so that the repl is kinda back to normal. Increase font size, add a total page background color(which doesn't cover the special elements), fix the timeline(use w3school's tutorial as that is what I used in the first place), fix the Repl.it 2020 thing coming on the posts page, such. Comment more on this in the other pull request. @BD103 Please see this and help fix these changes. These are the things I wanted to do today, but I couldn't finish them. You guys do them, **do not merge** and I won't merge tomorrow either. When I tell to merge on the pr. Merge tomorrow.

ColePete commented:
@BD103 can I help with anything?

darkdarcool commented:
Cool. I also added time if you scroll up on the main page

whippingdot commented:
> @BD103 can I help with anything?

Same thing

darkdarcool commented:
@whippingdot , I updated the repl auth and the js. Someone else will have to go and update the html to fit the site, but I did what was asked.

BD103 commented:
> @whippingdot , I updated the repl auth and the js. Someone else will have to go and update the html to fit the site, but I did what was asked.

Uh... I think I broke it.

whippingdot commented:
Did you fix it by now?

BD103 commented:
> Did you fix it by now?

No. I have no clue what I'm doing.

Error:
```
(node:219) UnhandledPromiseRejectionWarning: ReferenceError: fs is not defined
    at /home/runner/Replit-2021/index.js:15:11
    at Layer.handle [as handle_request] (/home/runner/Replit-2021/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/runner/Replit-2021/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/home/runner/Replit-2021/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/home/runner/Replit-2021/node_modules/express/lib/router/layer.js:95:5)
    at /home/runner/Replit-2021/node_modules/express/lib/router/index.js:281:22
    at Function.process_params (/home/runner/Replit-2021/node_modules/express/lib/router/index.js:335:12)
    at next (/home/runner/Replit-2021/node_modules/express/lib/router/index.js:275:10)
    at expressInit (/home/runner/Replit-2021/node_modules/express/lib/middleware/init.js:40:5)
    at Layer.handle [as handle_request] (/home/runner/Replit-2021/node_modules/express/lib/router/layer.js:95:5)
(node:219) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:219) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

whippingdot commented:
> > Did you fix it by now?
> 
> No. I have no clue what I'm doing.
> 
> Error:
> 
> ```
> (node:219) UnhandledPromiseRejectionWarning: ReferenceError: fs is not defined
>     at /home/runner/Replit-2021/index.js:15:11
>     at Layer.handle [as handle_request] (/home/runner/Replit-2021/node_modules/express/lib/router/layer.js:95:5)
>     at next (/home/runner/Replit-2021/node_modules/express/lib/router/route.js:137:13)
>     at Route.dispatch (/home/runner/Replit-2021/node_modules/express/lib/router/route.js:112:3)
>     at Layer.handle [as handle_request] (/home/runner/Replit-2021/node_modules/express/lib/router/layer.js:95:5)
>     at /home/runner/Replit-2021/node_modules/express/lib/router/index.js:281:22
>     at Function.process_params (/home/runner/Replit-2021/node_modules/express/lib/router/index.js:335:12)
>     at next (/home/runner/Replit-2021/node_modules/express/lib/router/index.js:275:10)
>     at expressInit (/home/runner/Replit-2021/node_modules/express/lib/middleware/init.js:40:5)
>     at Layer.handle [as handle_request] (/home/runner/Replit-2021/node_modules/express/lib/router/layer.js:95:5)
> (node:219) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
> (node:219) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
> ```

Come to our repl, I don't think our repl has the same problem...

whippingdot commented:
@BD103 actually...nope, it doesn't work for us too. Maybe delete the branch or discard all latest changes. You can copy the things you need and transfer it to a new branch with all the new features from the feature branch and you can make the changes. First we will try to fix this error tho...

BD103 commented:
> Come to our repl, I don't think our repl has the same problem...

That's the thing though. It's not on the repl because its the #6 branch, not Feature. I switched it back after having an error.

whippingdot commented:
> > Come to our repl, I don't think our repl has the same problem...
> 
> That's the thing though. It's not on the repl because its the #6 branch, not Feature. I switched it back after having an error.

No, I switched to your repl itself. It is broken.

darkdarcool commented:
I'm on know, what do I do today?

whippingdot commented:
Same things as before, but with one new thing. The style branch is **broken** so you need to fix it. There is some JS error so I need you to help fix it. Other than that the same things as I said yesterday:
> We are editing the branch that @BD103 created. Make it so that the repl is kinda back to normal. Increase font size, add a total page background color(which doesn't cover the special elements), fix the timeline(use w3school's tutorial as that is what I used in the first place), fix the Repl.it 2020 thing coming on the posts page, such. Comment more on this in the other pull request. @BD103 Please see this and help fix these changes. These are the things I wanted to do today, but I couldn't finish them. You guys do them, do not merge and I won't merge tomorrow either. When I tell to merge on the pr. Merge tomorrow.

Also check this out: https://dev.to/github/updates-to-github-markdown-making-it-easier-for-you-to-build-amazing-things-49fj

darkdarcool commented:
Alright I'm heading over there. Are you on the repl?

whippingdot commented:
Yes but most likely I won't be editing

darkdarcool commented:
KK

darkdarcool commented:
Super sorry, I couldn't figure out what was wrong with the node.js.

BD103 commented:
Well, @darkdarcool it seems we have three errors:
```
(node:219) UnhandledPromiseRejectionWarning: ReferenceError: fs is not defined
(node:219) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:219) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

So we need to define `fs`, add a `.catch()` / remove `async` key, and handle the promise rejections (whatever that is. you may just have to define fs, that might work)

darkdarcool commented:
Cool. I guess I did make some progress then.

firefish111 commented:
# STOP THE NOTIFICATIONS REEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE

AmazingMech2418 reviewed (approved):
Looks good. Just make sure again that it is the latest working version of the repl before merging.

whippingdot commented:
Wait, why are we commenting about the other pr here? This is the wrong pr, go on comment and approve the changes of the other pr.

whippingdot commented:
Umm I know this is random and totally off topic but @programmeruser2 how do you find all these repos on Github. You always star like 20 repos per day. Where do you find these repos?

darkdarcool commented:
I am not him, but it's on the explore tab in the main menu?

whippingdot commented:
> I am not him, but it's on the explore tab in the main menu?

There aren't that many, max 20 per user I think.

darkdarcool commented:
If you look at the menu in the explore tab, there are more things to look at.

darkdarcool commented:
I forgot to asl, I'm on now(duh), what am I supposed to do today?

whippingdot commented:
> I forgot to ask, I'm on now(duh), what am I supposed to do today?

Finish fixing the other branch. After that we share these CHANGES! P.S. Suggestions are only put in the console so please @AmazingMech2418 @tsunami-or-something @tickedoffsquid @darkdarcool @BD103 @ColePete WORK ON MAKING THEM GO TO A FILE. A txt file btw. PLEASE DO THIS TODAY. 😢 😭 😭 😭 😭 🥺 🥺 🙏

whippingdot commented:
@Isaiah08-D general commit meaning?

darkdarcool commented:
If II'm to do that, we need to stop reverting the repl back to an old on, the suggestion tab keeps disappearing

Isaiah08-D commented:
> @Isaiah08-D general commit meaning?
@whippingdot I didn't commit. That wasn't me.

whippingdot commented:
> If II'm to do that, we need to stop reverting the repl back to an old on, the suggestion tab keeps disappearing

Hmm...If that happens again, tell me

darkdarcool commented:
@whippingdot, do we really need the users suggestions to go to a txt file? I have been trying to get it to work but can't, and is it really that important. People will probably just goof off with it.

whippingdot commented:
> @whippingdot, do we really need the users suggestions to go to a txt file? I have been trying to get it to work but can't, and is it really that important. People will probably just goof off with it.

Then why in the world do we have a suggestions page?

darkdarcool commented:
I don't know, I wasn't the one to create it.

darkdarcool commented:
Wahoo! It's out. After months of work! It's time to celebrate!!
(and we can finally close this pull request)

whippingdot commented:
Lol ~~but we didn't even finish it properly~~ WE DONE FREEDOM

Next is probs a proj just for ourselves. We will share it in repl talk too as it will have our portfolios. This project was mentioned by both @darkdarcool and @Isaiah08-D. It will be a project starring each and every person in our org's profile. This will be the about page for this org. There will be a navbar from which you can access the portfolios.  The home page will just talk about the org and have the about, ~~with maybe my portfolio there cause I don't know how to make one **yet**~~ and then we will have a navbar which will have the links to everyone else's. Even in the footer we will have the other people's. We will also have a projects page and a secret page which is just random. We shud start working on this soon(as this project is WAY more complex then the 2020/2021 thing) so @AmazingMech2418, @BD103, @darkdarcool plan something for this. @BD103 u do css cause you hate mine, @darkdarcool do whatever you want(mostly the code), and @AmazingMech2418 like your mod perms try planning out the whole project. If you are not active on github and you don't see this it is ok as I will help, and @firefish111, @JBYT27, and @Codemonkey51 will all help too. @tsunami-or-something can also help with css, as so can @ColePete, @CodingRedpanda(previously ridark) and @CyanCoding(the 'C's). Everyone will help with the code too. Also if you want to be the main guy for something reply to this with what. I will copy this to the community discussions thing too. The three names at the start are the main people for their respective stuff(@AmazingMech2418 for planning, @darkdarcool for coding, and @BD103 for css and putting everyone under control). BOI BOI NOW ~~i have exams i have to study will be free this weekend~~

darkdarcool commented:
Hmm, would you do it in the node.js/html, or react.js? Or something completely different. I can probably get the skeleton for the project done.

whippingdot commented:
node.js/html it is the most popular and on this proj we need lots of people

darkdarcool commented:
Hmm, ok. I'll get the skeleton up.
